File: 2390. Removing Stars From a String/main.c
#include <stdbool.h>
#include <stdio.h>

char *removeStars(char *s) {
    size_t len = strlen(s);
    char *result = (char *)malloc((len + 1) * sizeof(char));

    if (result == NULL) {
        return NULL;
    }

    int count = 0;

    for (size_t i = 0; i < len; i++) {
        if (s[i] == '*') {
            count--;
        } else {
            result[count++] = s[i];
        }
    }

    result[count] = '\0';

    return result;
}

File: 2390. Removing Stars From a String/main.py
class Solution:
    def removeStars(self, s: str) -> str:
        result = ""

        for c in s:
            if c == "*":
                result = result[:-1]
            else:
                result += c

        return result


class Solution:
    def removeStars(self, s: str) -> str:
        result = []

        for c in s:
            if c == "*":
                result.pop()
            else:
                result.append(c)

        return "".join(result)
